### PR TITLE
Cleaning metric processes and adding additional checks

### PR DESCRIPTION
**Pull request by loumalouomega**

### CODE REVIEW

File: applications/MeshingApplication/custom_processes/metrics_hessian_process.cpp [hunk 1]
```diff
@@ -109,32 +102,23 @@ void ComputeHessianSolMetricProcess<TDim, TVarType>::Execute()
...
     
         Vector& metric = it_node->GetValue(MMG_METRIC);
         
-    #ifdef KRATOS_DEBUG 
-        if(metric.size() != TDim * 3 - 3) 
-        {
-            KRATOS_ERROR << "Wrong size of vector MMG_METRIC found for node " << it_node->Id() << " size is " << metric.size() << " expected size was " << TDim * 3 - 3;
-        }
-    #endif
+#ifdef KRATOS_DEBUG
```
Comment (rubenzorrilla) [inline applications/MeshingApplication/custom_processes/metrics_hessian_process.cpp:112]: Why you always check some variables and other variable are only checked in debug mode?

File: applications/MeshingApplication/custom_processes/metrics_hessian_process.cpp [hunk 2]
```diff
@@ -77,21 +74,17 @@ void ComputeHessianSolMetricProcess<TDim, TVarType>::Execute()
...
         auto it_node = nodes_array.begin() + i;
         
-    #ifdef KRATOS_DEBUG 
-        if ( it_node->SolutionStepsDataHas( mVariable ) == false )
-        {
-            KRATOS_ERROR << "Missing variable on node " << it_node->Id() << std::endl;
-        }
-    #endif
+        // Some checks
+        KRATOS_ERROR_IF_NOT(it_node->SolutionStepsDataHas( mVariable )) << "ERROR:: Missing variable on node " << it_node->Id() << std::endl;
```
Comment (rubenzorrilla) [inline applications/MeshingApplication/custom_processes/metrics_hessian_process.cpp]: There's a method in the VariablesUtils that performs this kind of check.

File: applications/MeshingApplication/custom_processes/metrics_levelset_process.cpp [hunk 3]
```diff
@@ -65,66 +62,51 @@ void ComputeLevelSetSolMetricProcess<TDim>::Execute()
...
-    {
+    for(int i = 0; i < num_nodes; ++i)  {
         auto it_node = nodes_array.begin() + i;
         
-        if ( it_node->SolutionStepsDataHas( mVariableGradient ) == false )
-        {
-            KRATOS_ERROR << "Missing gradient variable on node " << it_node->Id() << std::endl;
-        }
+        // Some checks
+        KRATOS_ERROR_IF_NOT(it_node->SolutionStepsDataHas( mVariableGradient )) << "ERROR:: Missing gradient variable on node " << it_node->Id() << std::endl;
```
Comment (rubenzorrilla) [inline applications/MeshingApplication/custom_processes/metrics_levelset_process.cpp]: Same as above, see the VariablesUtils().

File: applications/MeshingApplication/custom_processes/metrics_hessian_process.cpp (hunk 2 above)
Comment (rubenzorrilla) [reply applications/MeshingApplication/custom_processes/metrics_hessian_process.cpp]: If you want to check a unique node:
~~~
    const auto& r_node = *mrModelPart.NodesBegin();
    KRATOS_CHECK_VARIABLE_IN_NODAL_DATA(NODAL_H, r_node);
    KRATOS_CHECK_VARIABLE_IN_NODAL_DATA(DISTANCE, r_node);
~~~

**Comment by loumalouomega** [comment]
Ping

**Comment by loumalouomega** [comment]
Reping

**Comment by loumalouomega** [comment]
Ping

**Review by rubenzorrilla — APPROVED** [review]

**Pull request merged**